### DISCUSSION TITLE
How to overwrite the constants

### DISCUSSION THREAD

**Discussion by Xavier-M** [maxGraph/maxGraph#111] [Q&A]
Hi,

I'm importing maxgraph into my own development and I would like to overwrite some constants from the class Constants, ie:
HANDLE_SIZE, VERTEX_SELECTION_COLOR, GUIDE_COLOR, ...

It was possible with mxGraph but now they are real constants and it is not possible :-)

Is there a way to do this or should I override all functions using them? Thank you :)

**Comment by tbouffard** [comment]
Hi,
currently, there is no direct way to override the constants but in some cases, you can override the code that use them.

For instance, `VERTEX_SELECTION_COLOR` is only used as a return value of `VertexHandler.getSelectionColor`
What can be done is to extend the `VertexHandler` class and override the `getSelectionColor` method. Then, override the `Graph.createVertexHandler` method to return an instance of your custom class.
Alternatively, you can directly change the prototype of `VertexHandler.getSelectionColor` and return the value you want to use.

This may not work for every constant.
For instance, `HANDLE_SIZE` is used in various methods in `VertexHandler` so we aren't able to replicate the workaround proposed above except if we accept to override a lot of code.

So I suggest you to fill an issue for blocking cases. If you have any ideas or wishes about how to solve the issue, don't hesitate to share your thoughts 😸
